tests/data-url.test.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const dataUrlTransformer = require('../addons/with-web/data-url');
+const fs = require('fs');
+
+it('should return string containing data-url', () => {
+    const fileContent = 'this is a standard text file!';
+    const fileName = `${__dirname}/test-files/dataurl_test_file.txt`;
+
+    // Can be verified at https://regex101.com/r/r8tE5u/3
+    const matcher = /data:([\w/+]+);(charset=[\w-]+|base64),([a-zA-Z0-9+/]+={0,2})/;
+
+    fs.writeFileSync(fileName, fileContent);
+
+    const output = dataUrlTransformer.process(null, fileName);
+
+    fs.unlinkSync(fileName);
```

Comment:
What if we use https://github.com/kaelzhang/node-tmp-sync or similar to create us temporary directories that will be automatically removed? Advantages:

- No need to cleanup the files
- No need to change the gitignore

Reply:
This has been implemented, but i had to use tmp.clean() manually because i haven't been able to reproduce the automatic cleaning that the documentation mentions. Not sure what the problem is, but process.on('exit', ...) doesn't fire at all. However, i haven't yet figured out if this is a problem with my machine, Jest, Node, etc. 

Will continue to investigate, but, as a precaution, the commit still has tmp.clean(), though i presume this to be superfluous in most use cases.

Reply:
I think it has to do with the fact that Jest spawns multiple worker processes, one per test file. Let’s keep the cleanup explicitly though.